### PR TITLE
New resource: aws_apigatewayv2_integration

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Another resource for #7004.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
FEATURES:

* New Resource: aws_apigatewayv2_integration
```

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGateway2Integration_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSAPIGateway2Integration_ -timeout 120m
=== RUN   TestAccAWSAPIGateway2Integration_basic
=== PAUSE TestAccAWSAPIGateway2Integration_basic
=== RUN   TestAccAWSAPIGateway2Integration_IntegrationTypeHttp
=== PAUSE TestAccAWSAPIGateway2Integration_IntegrationTypeHttp
=== RUN   TestAccAWSAPIGateway2Integration_Lambda
=== PAUSE TestAccAWSAPIGateway2Integration_Lambda
=== RUN   TestAccAWSAPIGateway2Integration_VpcLink
=== PAUSE TestAccAWSAPIGateway2Integration_VpcLink
=== CONT  TestAccAWSAPIGateway2Integration_basic
=== CONT  TestAccAWSAPIGateway2Integration_VpcLink
=== CONT  TestAccAWSAPIGateway2Integration_IntegrationTypeHttp
=== CONT  TestAccAWSAPIGateway2Integration_Lambda
--- PASS: TestAccAWSAPIGateway2Integration_basic (23.83s)
--- PASS: TestAccAWSAPIGateway2Integration_IntegrationTypeHttp (39.78s)
--- PASS: TestAccAWSAPIGateway2Integration_Lambda (41.64s)
--- PASS: TestAccAWSAPIGateway2Integration_VpcLink (708.94s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	709.004s
```

~Currently based on https://github.com/terraform-providers/terraform-provider-aws/pull/8842, will need to rebase once that PR is merged.~

### CODE REVIEW

**Comment by ewbankkit** [comment]
Rebased to track https://github.com/terraform-providers/terraform-provider-aws/pull/8842#issuecomment-507334209.

**Comment by ewbankkit** [comment]
Rebased to fix merge conflict.
Acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGateway2Integration_'==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSAPIGateway2Integration_ -timeout 120m
=== RUN   TestAccAWSAPIGateway2Integration_basic
=== PAUSE TestAccAWSAPIGateway2Integration_basic
=== RUN   TestAccAWSAPIGateway2Integration_IntegrationTypeHttp
=== PAUSE TestAccAWSAPIGateway2Integration_IntegrationTypeHttp
=== RUN   TestAccAWSAPIGateway2Integration_Lambda
=== PAUSE TestAccAWSAPIGateway2Integration_Lambda
=== RUN   TestAccAWSAPIGateway2Integration_VpcLink
=== PAUSE TestAccAWSAPIGateway2Integration_VpcLink
=== CONT  TestAccAWSAPIGateway2Integration_basic
=== CONT  TestAccAWSAPIGateway2Integration_VpcLink
=== CONT  TestAccAWSAPIGateway2Integration_Lambda
=== CONT  TestAccAWSAPIGateway2Integration_IntegrationTypeHttp
--- PASS: TestAccAWSAPIGateway2Integration_basic (21.63s)
--- PASS: TestAccAWSAPIGateway2Integration_IntegrationTypeHttp (34.45s)
--- PASS: TestAccAWSAPIGateway2Integration_Lambda (38.25s)
--- PASS: TestAccAWSAPIGateway2Integration_VpcLink (722.33s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	722.386s
```

**Comment by ewbankkit** [comment]
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGateway2Integration_' 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSAPIGateway2Integration_ -timeout 120m
=== RUN   TestAccAWSAPIGateway2Integration_basic
=== PAUSE TestAccAWSAPIGateway2Integration_basic
=== RUN   TestAccAWSAPIGateway2Integration_IntegrationTypeHttp
=== PAUSE TestAccAWSAPIGateway2Integration_IntegrationTypeHttp
=== RUN   TestAccAWSAPIGateway2Integration_Lambda
=== PAUSE TestAccAWSAPIGateway2Integration_Lambda
=== RUN   TestAccAWSAPIGateway2Integration_VpcLink
=== PAUSE TestAccAWSAPIGateway2Integration_VpcLink
=== CONT  TestAccAWSAPIGateway2Integration_basic
=== CONT  TestAccAWSAPIGateway2Integration_VpcLink
=== CONT  TestAccAWSAPIGateway2Integration_Lambda
=== CONT  TestAccAWSAPIGateway2Integration_IntegrationTypeHttp
--- PASS: TestAccAWSAPIGateway2Integration_basic (22.25s)
--- PASS: TestAccAWSAPIGateway2Integration_IntegrationTypeHttp (34.91s)
--- PASS: TestAccAWSAPIGateway2Integration_Lambda (38.44s)
--- PASS: TestAccAWSAPIGateway2Integration_VpcLink (734.10s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	734.131s
```

**Comment by ewbankkit** [comment]
After merge of https://github.com/terraform-providers/terraform-provider-aws/pull/8842, incorporate all the feedback:
* Change resource name to `aws_apigatewayv2_blah`; Rename source files
* Add `subcategory: "API Gateway v2 (WebSocket and HTTP APIs)"` to documentation
* Change function names to `ApiGatewayV2`
* Add `_disappears` acceptance test
* Use `multierror` in any test sweeper

**Comment by ewbankkit** [comment]
Renamed resource to `aws_apigatewayv2_integration`.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Integration_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Integration_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Integration_basic
=== PAUSE TestAccAWSAPIGatewayV2Integration_basic
=== RUN   TestAccAWSAPIGatewayV2Integration_disappears
=== PAUSE TestAccAWSAPIGatewayV2Integration_disappears
=== RUN   TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== RUN   TestAccAWSAPIGatewayV2Integration_Lambda
=== PAUSE TestAccAWSAPIGatewayV2Integration_Lambda
=== RUN   TestAccAWSAPIGatewayV2Integration_VpcLink
=== PAUSE TestAccAWSAPIGatewayV2Integration_VpcLink
=== CONT  TestAccAWSAPIGatewayV2Integration_basic
=== CONT  TestAccAWSAPIGatewayV2Integration_Lambda
=== CONT  TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_disappears
=== CONT  TestAccAWSAPIGatewayV2Integration_VpcLink
--- PASS: TestAccAWSAPIGatewayV2Integration_disappears (23.85s)
--- PASS: TestAccAWSAPIGatewayV2Integration_basic (27.84s)
--- PASS: TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp (43.88s)
--- PASS: TestAccAWSAPIGatewayV2Integration_Lambda (52.86s)
--- PASS: TestAccAWSAPIGatewayV2Integration_VpcLink (657.77s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	657.827s
```

**Comment by ewbankkit** [comment]
This is now ready for review.

File: aws/resource_aws_apigatewayv2_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+}
+`, rName)
+}
+
+func testAccAWSAPIGatewayV2IntegrationConfig_vpcLink(rName string) string {
+	return testAccAWSAPIGatewayV2IntegrationConfig_apiWebSocket(rName) + fmt.Sprintf(`
+data "aws_availability_zones" "available" {
+  state = "available"
+
+// https://github.com/terraform-providers/terraform-provider-aws/issues/12412
```
Comment (gdavison) [inline aws/resource_aws_apigatewayv2_integration_test.go]: #14200 has been merged now, so this is supported

Comment (ewbankkit) [reply aws/resource_aws_apigatewayv2_integration_test.go]: OK, thanks; I'll rebase, remove the comment and retest in `us-west-2` (my testing account now has the LAX1 local zone enabled).

Comment (gdavison) [reply aws/resource_aws_apigatewayv2_integration_test.go]: 👍

**Comment by ewbankkit** [comment]
Rebased after merge of https://github.com/terraform-providers/terraform-provider-aws/pull/12400.

Re-ran affected acceptance test:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Integration_VpcLink'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Integration_VpcLink -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Integration_VpcLink
=== PAUSE TestAccAWSAPIGatewayV2Integration_VpcLink
=== CONT  TestAccAWSAPIGatewayV2Integration_VpcLink
--- PASS: TestAccAWSAPIGatewayV2Integration_VpcLink (698.17s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	698.227s
```

**Comment by ewbankkit** [comment]
I will add a new `aws_apigatewayv2_vpc_link` as a separate PR and include additional `aws_apigatewayv2_integration` acceptance tests for [private integrations for HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-private.html) then.

**Pull request merged**